superset-frontend/src/dashboard/util/downloadAsImage.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { MouseEvent, MouseEventHandler } from 'react';
+import domToImage from 'dom-to-image';
+import kebabCase from 'lodash/kebabCase';
+
+/**
+ * generate a consistent file stem from a description and date
+ *
+ * @param description title or description of content of file
+ * @param date date when file was generated
+ */
+export const generateFileStem = (description: string, date = new Date()) => {
+  return `${kebabCase(description)}-${date
+    .toISOString()
+    .replace(/[: ]/g, '-')}`;
+};
+
+/**
+ * Create an event handler for turning an element into an image
+ *
+ * @param selector css selector of the parent element which should be turned into image
+ * @param fileStem name of generated file, without extension
+ * @param backgroundColor background color to apply to screenshot document
+ */
+export const downloadAsImage = (
+  selector: string,
+  fileStem: string,
+  backgroundColor = '#f5f5f5',
```

Comment:
can we pull this color out into a constant? Or better yet, is it defined somewhere in a theme file already as a constant to be reused here?

Reply:
I pulled this from https://github.com/apache/incubator-superset/blob/c53bc4ddf9808a8bb6916bbe3cb31935d33a2420/superset-frontend/stylesheets/less/variables.less#L34
Currently there is not connector pulling the less variables into JS.
I could create an adapter that tries to read the variable at runtime, or could create a new constants file, thoughts? :thought_balloon:

Reply:
I'd just do something like `const GRAY_BACKGROUND_COLOR = "#F5F5F5";` at the top of this file and use it as the default argument here. I've mentioned this issue with less variables not being in JS in another PR too, so i think we'll probably pull them out soon and then do a refactor throughout all the code (and thus clean up the usage here too).

Reply:
Sounds good, thanks!
updated